### PR TITLE
Remove historical features infrastructure

### PR DESCRIPTION
**Pull request by thecoop**
Remove all the infrastructure around historical features, we only need regular features now

### CODE REVIEW

**Review by mark-vieira — COMMENTED** [review]
I thought all the build time stuff was only necessary for "historical" features. Seems we're just renaming this to remove the "historical" aspect but build time generation is still required even for the new cluster features?

**Comment by thecoop** [comment]
We also record all the registered features so we can check for typos in the tests - this was added later, but the classes were not renamed

**Comment by thecoop** [comment]
Just waiting on #116966 to remove the ESQL features

**Comment by mark-vieira** [comment]
> We also record all the registered features so we can check for typos in the tests - this was added later, but the classes were not renamed

I'm wondering if we should re-evaluate whether all this complexity is worth it just so that if someone fat-fingers a feature name in a test they get a slightly better error. Or if there's a way we can still provide that without all this build-time complexity.

cc @rjernst

**Comment by thecoop** [comment]
It's not that they'll get an error, it's that the test won't run at all, with no notification - and they might not notice.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java [hunk 1]
```diff
@@ -557,9 +557,6 @@ public static Set<String> capabilities(boolean all) {
         for (NodeFeature feature : new EsqlFeatures().getFeatures()) {
             caps.add(cap(feature));
         }
-        for (NodeFeature feature : new EsqlFeatures().getHistoricalFeatures().keySet()) {
-            caps.add(cap(feature));
-        }
```
Comment (nik9000) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:562]: :+1:

File: x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java [hunk 2]
```diff
@@ -207,10 +206,7 @@ protected static void checkCapabilities(RestClient client, TestFeatureService te
             }
         }
 
-        var features = Stream.concat(
-            new EsqlFeatures().getFeatures().stream(),
-            new EsqlFeatures().getHistoricalFeatures().keySet().stream()
-        ).map(NodeFeature::id).collect(Collectors.toSet());
+        var features = new EsqlFeatures().getFeatures().stream().map(NodeFeature::id).collect(Collectors.toSet());
```
Comment (nik9000) [inline x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/EsqlSpecTestCase.java:209]: :+1:

**Comment by ldematte** [comment]
We have to have a way to distinguish between "feature not present" and "feature does not exists".
I'm in favor of living the current PR as-is, and think if there is an alternative after. It's true these are extra build steps, but we have them, so it's worth keeping them for now and thing about an alternative later (especially since an alternative is not immediately apparent - at least to me, I cannot think at a way that does not involve a build step)

**Review by ldematte — COMMENTED** [review]
The changes look good to me.
As I wrote above, I'm in favor of living the current PR as-is and following up if we find another way. If we are in agreement, I'm for merging it.

**Review by ldematte — APPROVED** [review]

**Review by mark-vieira — APPROVED** [review]

**Pull request merged**